daemon/src/payout_curve/csr_tools.rs

```diff
@@ -149,4 +148,61 @@ mod tests {
 
         assert!(matches!(a, Error::CannotInitCSR));
     }
+
+    // test that all is good
+    #[test]
+    fn test_lu_matrix_solve_00() {
+        let a = CSR::new(
+            Array1::<f64>::from_vec(vec![11., 12., 22., 23., 31., 33.]),
+            Array1::<usize>::from_vec(vec![0, 1, 1, 2, 0, 2]),
+            Array1::<usize>::from_vec(vec![0, 2, 4, 6]),
+            (3, 3),
+        )
+        .unwrap();
+        let b = Array2::<f64>::from(vec![
+            [36., 59., 82.],
+            [204., 249., 294.],
+            [198., 262., 326.],
+        ]);
+        let x_expected = Array2::<f64>::from(vec![[0., 1., 2.], [3., 4., 5.], [6., 7., 8.]]);
+        let x = a.matrix_solve(&b).unwrap();
+
+        assert!(x.abs_diff_eq(&x_expected, 1e-10));
+    }
+
+    // test that an indeterminate system borks
+    #[test]
+    fn test_lu_matrix_solve_01() {
+        let a = CSR::new(
+            Array1::<f64>::from_vec(vec![1.]),
+            Array1::<usize>::from_vec(vec![2]),
+            Array1::<usize>::from_vec(vec![0, 1, 1, 1]),
+            (3, 3),
+        )
+        .unwrap();
+        let b = Array2::<f64>::from(vec![
+            [36., 59., 82.],
+            [204., 249., 294.],
+            [198., 262., 326.],
+        ]);
+        let e = a.matrix_solve(&b).unwrap_err();
```

Comment:
I'd prefer a newline above this statement so that the test is visually grouped into:

ARRANGE

ACT

ASSERT